pkg/controller/workload/job/job_webhook.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package job
+
+import (
+	"context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+)
+
+// log is for logging in this package.
+var webhookLog = ctrl.Log.WithName("job-webhook")
+
+type JobWebhook struct {
+	manageJobsWithoutQueueName bool
+}
+
+// SetupWebhook configures the webhook for batchJob.
+func SetupWebhook(mgr ctrl.Manager, opts ...Option) error {
+	options := defaultOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	wh := &JobWebhook{
+		manageJobsWithoutQueueName: options.manageJobsWithoutQueueName,
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&batchv1.Job{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-batch-v1-job,mutating=true,failurePolicy=fail,sideEffects=None,groups=batch,resources=jobs,verbs=create,versions=v1,name=mjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &JobWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *JobWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*batchv1.Job)
+	webhookLog.V(5).Info("Applying defaults", "job", klog.KObj(job))
```

Comment:
I don't think we did this in the other webhooks, but did you try getting a logger from the context?

In general, using a logger from the context allows to include, well, contextual information that might be coming from the callers of the function.

Reply:
This is the general pattern we follow in other places, see https://github.com/kubernetes-sigs/kueue/blob/f24c63accaad461dfe582b21819dbf3a5d75dd60/apis/kueue/webhooks/workload_webhook.go#L37-L56

Reply:
I know.... but I think that was by mistake.

Reply:
updated